pandas/tests/reshape/test_pivot.py

```diff
@@ -65,6 +65,25 @@ def test_pivot_table(self):
             index + [columns])['D'].agg(np.mean).unstack()
         tm.assert_frame_equal(table, expected)
 
+    def test_pivot_table_categorical_observed(self, observed):
+        # issue #24923
+        df = pd.DataFrame({'col1': list('abcde'),
+                           'col2': list('fghij'),
+                           'col3': [1, 2, 3, 4, 5]})
+
+        df.col1 = df.col1.astype('category')
+        df.col2 = df.col1.astype('category')
+
+        expected = df.pivot_table(index='col1', values='col3',
```

Comment:
Sorry just on last check - am I reading this right? If this matches the result wouldn't that mean that the observed argument isn't actually doing anything?

Reply:
Apologies if I'm missing something obvious here, it's been a while since I last looked at this pull request. 

I thought this was a valid test as it checks the faster instance (where observed is passed as True and cartesian cross products are avoided) with an instance where it defaults to False - ensuring that the outputs are equivalent, which they should be.

Reply:
Ah OK makes sense in that regards. Aren't we missing a test where `observed=True` and `observed=False` would actually make a difference in the result though?

Reply:
As far as I'm aware, it makes use of the pytest fixture for observed.

When I execute the following test individually:

`pytest -s -v pandas/tests/reshape/test_pivot.py`

You see the test pass with all three scenarios.

`pandas/tests/reshape/test_pivot.py::TestPivotTable::test_pivot_table_categorical_observed[True] PASSED
pandas/tests/reshape/test_pivot.py::TestPivotTable::test_pivot_table_categorical_observed[False] PASSED
pandas/tests/reshape/test_pivot.py::TestPivotTable::test_pivot_table_categorical_observed[None] PASSED`